src/client.ts

```diff
@@ -263,6 +265,34 @@ class OptimizelyReactSDKClient implements ReactSDKClient {
     });
   }
 
+  getUserContextInstance(userInfo: UserInfo): optimizely.OptimizelyUserContext | null {
+    let userContext: optimizely.OptimizelyUserContext | null = null;
+
+    const userId = userInfo.id || this.user.id;
+    const userAttributes = userInfo.attributes || this.user.attributes;
+
+    if (this.userContext) {
+      if (JSON.stringify(userInfo) === JSON.stringify(this.user)) {
+        return this.userContext;
+      }
+
+      if (userId) {
+        userContext = this._client.createUserContext(userId, userAttributes);
+        return userContext;
+      }
+
+      return null;
```

Comment:
If you simply remove this whole part, it will be covered by next code block because both are essentially doing the exact same thing.

Reply:
Ignore this comment! I understood it wrong. my bad.